tests/components/application-connector/test/application-connectivity-validator/suite_test.go

```diff
@@ -0,0 +1,133 @@
+package application_connectivity_validator
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/kyma-project/kyma/tests/components/application-connector/internal/testkit/httpd"
+)
+
+const v1EventsFormat = "http://central-application-connectivity-validator.kyma-system:8080/%s/v1/events"
+const v2EventsFormat = "http://central-application-connectivity-validator.kyma-system:8080/%s/v2/events"
+
+const XForwardedClientCertFormat = "Hash=hash1;Cert=\"cert\";Subject=\"O=client organization,CN=%s\";URI=,By=spiffe://cluster.local/ns/default/sa/echoserver;Hash=hash;Subject=\"\";URI=spiffe://cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account"
+
+const standaloneAppName = "event-test-standalone"
+const compassAppName = "event-test-compass"
+
+type ValidatorSuite struct {
+	suite.Suite
+}
+
+func (vs *ValidatorSuite) SetupSuite() {
+}
+
+func (vs *ValidatorSuite) TearDownSuite() {
+	_, err := http.Post("http://localhost:15000/quitquitquit", "", nil)
+	vs.Nil(err)
+	_, err = http.Post("http://localhost:15020/quitquitquit", "", nil)
+	vs.Nil(err)
+}
+
+func TestValidatorSuite(t *testing.T) {
+	suite.Run(t, new(ValidatorSuite))
+}
+
+func (vs *ValidatorSuite) TestGoodCert() {
+	cli := httpd.NewCli(vs.T())
+
+	for _, testCase := range []struct {
+		appName       string
+		expectedCName string
+	}{{
+		appName:       standaloneAppName,
+		expectedCName: standaloneAppName,
+	}, {
+		appName:       compassAppName,
+		expectedCName: "clientId1",
+	}} {
+		v1Events := fmt.Sprintf(v1EventsFormat, testCase.appName)
+		v2Events := fmt.Sprintf(v2EventsFormat, testCase.appName)
+		endpoints := []string{v1Events, v2Events}
```

Comment:
I think we should add here a call to another endpoint `/events` which is handled in application-connectivity-validator  in mapRequestToProxy function. This endpoint should proxy event to `/publish` endpoint of event-publisher-proxy.  See https://github.com/kyma-project/kyma/blob/main/components/central-application-connectivity-validator/internal/validationproxy/handler.go#L125

Reply:
Please also check unit tests for mapping in application validator component. Maybe we need to write more test cases for checking the impact of valid/invalid of group and tenant fields in cert info header. Ex. `OU=group,O=tenant` But I also can assume this might be a dead code and there is no point for that. 

 https://github.com/kyma-project/kyma/blob/c3289d456b295485049f2ca51a9156f7d9cd7260/components/central-application-connectivity-validator/internal/validationproxy/handler_test.go#L71

Reply:
> Please also check unit tests for mapping in application validator component. Maybe we need to write more test cases for checking the impact of valid/invalid of group and tenant fields in cert info header. Ex. `OU=group,O=tenant` But I also can assume this might be a dead code and there is no point for that.
> 
> https://github.com/kyma-project/kyma/blob/c3289d456b295485049f2ca51a9156f7d9cd7260/components/central-application-connectivity-validator/internal/validationproxy/handler_test.go#L71

These fields in cert are not impacting the validity of our tests. This is an obsolete feature.